### ISSUE TITLE
Too many /tmp/go-buildXXXX directories

### ISSUE THREAD

**Issue by StephenPCG** [fatih/vim-go#1249]
### Behavior

vim-go produces too many ``/tmp/go-buildXXXX`` directories.

### Steps to reproduce:

Activate vim-go using pathogen. Only have the following configuration for vim-go:

```
  let g:go_bin_path = g:vimrcroot . "cache/vim-go/"
  au FileType go nmap K <Plug>(go-doc)
  au FileType go nmap gd <Plug>(go-def-tab)
```

1. Use ``watch`` command to monitor number of ``/tmp/go-build*`` directories:
```
watch -n1 'ls -1d /tmp/go-build* | wc -l'
```

2. Open a go file with vim, save multiple times as quickly as possible.  Number of ``/tmp/go-buildXXX`` directories grows.

I guess that, on file save, vim-go forks some ``go`` commands, which create the tmp directories and delete them on exit. But if I trigger another save before such commands exit, vim-go would kill running commands, which prevents the tmp directories from being deleted.

I believe there are other occasions which will cause such tmp directories not be deleted. I opened a vim for less than two weeks, and there were more than 50 thousand such tmp directories.

### Version in use

This issue has been there for at least 3 months, and currently still exist with master branch.

**Comment by fatih** [comment]
HI @StephenPCG 

I think you're using a different plugin. We don't build anything on save, it must be something else. Can you please remove all plugins and only use vim-go? You'll see the difference then.

**Comment by StephenPCG** [comment]
I did some more test, it seems that it's not vim-go's problem. I post the behavior below in case some others come across the same problem.

I disabled all plugins, and then:
* enable just vim-go, everything is ok
* enable just [ale](https://github.com/w0rp/ale), everything is ok
* enable both vim-go and ale, the problem appears

So it seems there are some kind of conflict between vim-go and ale, but I don't know who to blame.
